File: C#.Net/Tests/JDI UIWebTests/JDI UIWebTests/UIObjects/Pages/DatesPage.cs
using Epam.JDI.Core.Interfaces.Common;
using JDI_Web.Attributes;
using JDI_Web.Selenium.Elements.Common;
using JDI_Web.Selenium.Elements.Composite;
using JDIWebTests.UIObjects.Sections;

namespace JDI_UIWebTests.UIObjects.Pages
{
    public class DatesPage:WebPage
    {
        [FindBy(Css = "#datepicker input")]
        public DatePicker Datepicker;

        [FindBy(Css = "[data-provides=fileinput]")]
        public IFileInput ImageInput;

        [FindBy(Css = "")]
        public ILabel UploadedFileName;

        [FindBy(Css = "main form")]
        public ContactForm ContactForm;
    }
}

File: Java/Tests/jdi-uitest-webtests/src/main/java/com/epam/jdi/uitests/testing/unittests/pageobjects/pages/DatesPage.java
package com.epam.jdi.uitests.testing.unittests.pageobjects.pages;

import com.epam.jdi.uitests.core.interfaces.common.IFileInput;
import com.epam.jdi.uitests.core.interfaces.common.ILabel;
import com.epam.jdi.uitests.web.robot.RFileInput;
import com.epam.jdi.uitests.web.selenium.elements.common.DatePicker;
import com.epam.jdi.uitests.web.selenium.elements.composite.WebPage;
import com.epam.jdi.uitests.web.selenium.elements.pageobjects.annotations.ImageFile;
import org.openqa.selenium.support.FindBy;

/**
 * Created by Maksim_Palchevskii on 8/17/2015.
 */
public class DatesPage extends WebPage {
    @ImageFile("datepicker2.png")
    @FindBy(css = "#datepicker input")
    public DatePicker datepicker;
    @FindBy(css = "input[type=file]")
    public IFileInput imageInput;
    @FindBy(css = "[data-provides=fileinput]")
    public RFileInput rImageInput;
    @FindBy(css = "[class=filename] span")
    public ILabel uploadedFileName;
    @FindBy(css = "main form")
    public ContactForm contactForm;
}
